### PR TITLE
Integration tests for log4net forwarding

### LINKED ISSUE #909: IntegrationTests for Log Forwarding - log4net
### Feature Description

----

**IMPORTANT**
Until PR #894 is merged, base branches off of `log-metrics` branch.
Merge any PRs into `logsisnt-feature-branch` branch

----

- Confirm logs are sent when enabled, not when disabled
- Confirm that log counts follow max samples (set a low amount)
  - Will need to either add a new method to setup the log config or do it manually (did it manually for log metrics)
- Confirm that supportability metrics are created with correct counts

Acceptance Criteria

- Tests pass

### PR DESCRIPTION
**Pull request by JcolemanNR**
## Description

Closes #909

This adds integration test coverage for the log4net metrics and log forwarding functionality. It does not exhaustively test every scenario, but does cover happy path use cases, and likely customer configurations.

Several issues were identified while writing these tests:
- We do not support version 2.0.0 - 2.0.5 of log4net. (note: these versions are only available for .net framework)
- The MaxSamplesStored functionality will not send ANY data when low values are configured. The tests currently target a max samples stored of 1 and nothing gets sent.

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [x] Performance testing completed with satisfactory results (if required)
- [x] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

**Comment by jaffinito** [comment]
Looks like a great start!

File: src/Agent/NewRelic/Agent/Core/Aggregators/LogEventAggregator.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@
 using NewRelic.Agent.Core.Time;
 using NewRelic.Agent.Core.WireModels;
 using NewRelic.Collections;
+using NewRelic.Core.Logging;
```
Comment (jaffinito) [inline src/Agent/NewRelic/Agent/Core/Aggregators/LogEventAggregator.cs:14]: nitpicky: This using is not needed.

**Review by jaffinito — APPROVED** [review]
Awesome work!  Great finds on the bugs in the RecordLogMessage method.

File: src/Agent/NewRelic/Agent/Core/Aggregators/LogEventAggregator.cs (hunk 1 above)
Comment (JcolemanNR) [reply src/Agent/NewRelic/Agent/Core/Aggregators/LogEventAggregator.cs:14]: Aww dang! It's leftover from the debugging log message I added then removed. Good catch

**Review by nr-ahemsath — APPROVED** [review]
I did some refactoring to consolidate the testing of the four combinations of metrics enabled/disabled and forwarding enabled/disabled down to a single test file.  Looks good to me. 👍

**Pull request merged**